Tests/LiquidProjections.NHibernate.Specs/NHibernateProjectorSpecs.cs

```diff
@@ -1197,6 +1195,43 @@ private class ChildProjectionState
                 public bool Entry2Exists { get; set; }
             }
         }
+
+        public class When_event_handling_fails :
+            Given_a_sqlite_projector_with_an_in_memory_event_source
+        {
+            private const string originalExceptionMessage = "Some fail.";
+
+            public When_event_handling_fails()
+            {
+                Given(() =>
+                {
+                    Events.Map<CategoryDiscontinuedEvent>().As((@event, context) =>
+                    {
+                        throw new InvalidOperationException(originalExceptionMessage);
+                    });
+
+                    StartProjecting();
+                });
+            }
+
+            [Fact]
+            public async Task Then_it_should_throw_projection_exception()
+            {
+                ProjectionException exception = await Assert.ThrowsAsync<ProjectionException>(

```

Comment:
It's weird not to have a `When`. You can employ Chill's [result-type mechanism](https://github.com/dennisdoomen/VisionSuite/commit/2facc230c0569848e0d7dac969c89ebc2d7e2d01#diff-6541c4c2ab18df4686ce3262c414979cR216) to support this. Same for the RavenDB specs.